### DISCUSSION TITLE
[WalletConnectUnity] Failed to get wallets: HTTP/1.1 403 Forbidden

### DISCUSSION THREAD

**Discussion by Dilosen** [WalletConnect/WalletConnectUnity#217] [Q&A] [answered]
I am having an issue with getting wallet/s - not the chains, but the wallets themselves. I have verified my URL in the WalletConnect dashboard, and have an App ID for the project.  This worked out of the box; generating a QR, a list of wallets etc. However, after switching platform I am having an error of "forbidden" -

To Recreate:

1. download sample
2. switch platform to Android from default Windows, Mac 
3. add AppId from WalletConn Cloud
4. Run & Build out onto device OR
5. Run in Editor using Unity Remote as well

The issue is also logged using the logger provided in the script. I don't see any explict reason for denying the request - the logs show several successful connections and checks, for the blockchain and then for the wallets. The request is then sent, but the return is the 403 Forbidden error. This only happens once.

It then also logs after this, restarting the connection to the name of the App listed in the WalletConnect Dashboard; "Restarting transport for GAMES" etc. it also logs "Sending request irn_unsubscribe with data {"method":"irn_unsubscribe","params": ...." but not returning any message either. I assume it's trying to now disconnect.

I am not sure what is denying the request to the server, as the App ID was working prior to switching platform when I tried it a few hours ago.

FYI: logs for any selected chain;

JsonRpcProvider] Sending request irn_subscribe with data{...}
[WalletConnectUnity] Failed to get wallets: HTTP/1.1 403 Forbidden
 
OR

JsonRpcProvider] Sending request irn_publish with data {...}
[WalletConnectUnity] Failed to get wallets: HTTP/1.1 403 Forbidden

Edit: I switched platform back from Android to Windows, Mac and the issue persists. There is also another notable thing: The QR code does not render in the Android version, but in the Windows version it renders a QR code correctly. Not sure if this related to the 401 Forbidden request or something to do with the platform itself.

**Comment by skibitsky** [comment]
Hello,

This is a very strange issue. The WalletConnect Modal uses WebSockets for communication with the wallets through the Relay server and HTTP for loading wallet data, wallet images, and network images (the list of networks itself is hardcoded).

The most likely reason for receiving an `HTTP/1.1 403 Forbidden` error is that something is wrong with the project ID. However, in that case, WebSocket wouldn't work either, and your console would also include WebSocket errors.
<img width=700 src="https://github.com/WalletConnect/WalletConnectUnity/assets/15127974/104a0661-7c16-48cc-a14f-831345770ea4" />

But, as I understand, you are only getting an HTTP error. Our package uses `UnityWebRequest`, which is a wrapper around Unity's native HTTP client. Maybe platform switching broke it somehow. May I ask you to try restarting the Unity Editor, and if it doesn't help, deleting the `Library` directory?

> The QR code does not render in the Android version

On mobile platforms, we don't show the QR code on the first screen because it's rarely used on mobile devices. You can still access the QR code by navigating to `All wallets` → clicking on the `QR Code` icon next to the search bar.

Just in case, this is how mobile UI looks like:
<img width=350 src="https://github.com/WalletConnect/WalletConnectUnity/assets/15127974/4acdd818-026c-4906-ad89-f365dfe8d712" />

**Comment by skibitsky** [comment]
@moosesnWoop, could you please share your project ID? It's a public-facing indicator, safe to share it publicly.

**Comment by Cali93** [comment, marked as answer]
Hey @moosesnWoop,

Thanks for reporting the issue and your patience.
There was a bug on our Web3Modal API where all project IDs created from April 18 until earlier today were considered invalid.
The issue has been resolved and the requests should now get a valid response.

Sorry for the inconvenience.

**Reply by Dilosen** [reply]
Thanks all for your help! Indeed, I did create a new app ID when I came back to this project. So that probably was it.

I have just got back to my desk and tested it again on mobile. All seems to be working fine now as we can connect to Walletconnect and then get the list of available wallets etc.

Just a side note: I used the Algorand network chain, but in my ledger app it says the following: "Games requested to connect to chains we don't support yet" and so, I'm not sure if the Algorand chain is available yet?

![qrewq](https://github.com/WalletConnect/WalletConnectUnity/assets/16508683/348e6ff6-9056-475f-a9a0-340f23cffa0c)

**Reply by skibitsky** [reply]
It looks like Ledger Live wallet doesn't support Algorand. You should reach out to Ledger about that.

**Reply by Dilosen** [reply]
mm strange, it seems that Algorand is on Ledger. I'm not sure why but tbh doesn't matter if there are other available chains processing the same. Let me see if I can double check that algorand blockchain address.

I haven't tried metamask yet but also Pera wallet does not show up on my device in the modal. I will try via QR code, but that means it's a desktop build and not on the device. Not sure exactly how to call the Pera wallet from the modal without a QR code.

Edit: I've switched platform to PC and tried to connect to algorand using Pera wallet. I get the message that there is a mismatch, that the dapp is in testnet and the wallet in mainnet or visa versa.

I don't see an option in the sample that changes the network to testnets, but the chain selected in the modal should be algorand mainnet. (https://docs.walletconnect.com/advanced/multichain/chain-list 404's on me).

**Reply by skibitsky** [reply]
It's common for wallets to not support test chains, maybe Ledger Live works only with Algorand's main net?

> Pera wallet does not show up on my device in the modal

I don't see Pera wallet in WalletConnect Explorer. Do you know if they applied?

> https://docs.walletconnect.com/advanced/multichain/chain-list 404's on me

Apologies, it seems that chain list has a new URL:
https://docs.walletconnect.com/cloud/chains/chain-list

**Reply by Dilosen** [reply]
Correct, Ledger only works with mainnets, unlike metamask - but herein is where I had the error mismatch using the algorand blockchain out of the box.

So I did not know where "testnet" was actually coming from or how it was being detected - this was the same issue on Pera's wallet via QR code. The error is "there is a network mismatch. Either the dapp is on testnet or the wallet is on mainnet or visa versa"

So I puzzled a bit, added a testnet version of algorand, and then got the same result, "there is a network mismatch. Either the dapp is on testnet or the wallet is on mainnet or visa versa" .

If you see in the earlier screenshot, the working walletconnect sample of Algorand is not supported on Ledger for some reason, but Ledger has supported algorand chain for some time now. Could there be some network id mismatch here? surely not for the blockchain? I am getting this across all wallets for Algorand, not just Ledger.

**Reply by skibitsky** [reply]
Could you please open an issue?
